packages/cmf/scripts/cmf-settings.i18n.js

```diff
@@ -0,0 +1,255 @@
+const path = require('path');
+const fs = require('fs');
+const jsonpath = require('jsonpath');
+const cloneDeep = require('lodash/cloneDeep');
+const difference = require('lodash/difference');
+const get = require('lodash/get');
+const intersection = require('lodash/intersection');
+const set = require('lodash/set');
+const mkdirp = require('mkdirp');
+
+const { getLogger } = require('./cmf-settings.utils');
+const { getJSON } = require('./getJSON');
+
+const JSON_PATH_EXPRESSION = '$..i18n';
+const DEFAULT_LOCALE = 'en';
+
+/**
+ * getPathFromPattern - get the path from the current pattern
+ *
+ * @param  {string} pattern   is the template string of the path with {{namespace}} and {{local}}
+ * @param  {string} namespace namespace to set
+ * @param  {string} locale    local to set
+ * @return {string}           path with replaced value
+ */
+function getPathFromPattern(pattern, namespace, locale) {
+	const replaceMap = {
+		'{{namespace}}': namespace,
+		'{{locale}}': locale,
+	};
+
+	return path.join(
+		process.cwd(),
+		...pattern.replace(/{{namespace}}|{{locale}}/g, match => replaceMap[match]).split('/'),
+	);
+}
+
+/**
+ * getLocaleByNamespace - transform a JSON to a dictionary of key/value with a given namespace
+ *
+ * @param  {object} settings  JSON Object
+ * @param  {string} namespace namespace to parse
+ * @return {Map}              dictionary of key/value locales
+ */
+function getLocaleByNamespace(settings, namespace) {
+	return jsonpath
+		.query(settings, JSON_PATH_EXPRESSION)
+		.reduce(
+			(locale, i18n) => locale.set(i18n.key.split(`${namespace}:`)[1], i18n.options.defaultValue),
```

Comment:
this namespace make our translation very repetitive. We should put it with a default value in cmf.json and put it in a extra attribute. (i18n.domain)

Reply:
it's the syntax of i18n.t(key) where key is a string with a namespace and a key separated with a colon. e.g. namespace:key